src/lib/storage/chunk.cpp

```diff
@@ -17,14 +17,18 @@
 
 namespace opossum {
 
-Chunk::Chunk(const Segments& segments, const std::shared_ptr<MvccData>& mvcc_data,
+Chunk::Chunk(Segments segments, const std::shared_ptr<MvccData>& mvcc_data,
              const std::optional<PolymorphicAllocator<Chunk>>& alloc)
-    : _segments(segments), _mvcc_data(mvcc_data) {
+    : _segments(std::move(segments)), _mvcc_data(mvcc_data) {
 #if HYRISE_DEBUG
-  const auto chunk_size = segments.empty() ? 0u : segments[0]->size();
+  const auto chunk_size = _segments.empty() ? 0u : _segments[0]->size();
+  const auto is_reference_chunk = !_segments.empty() ? std::dynamic_pointer_cast<ReferenceSegment>(_segments.front()) != nullptr : false;
+
   Assert(!_mvcc_data || _mvcc_data->size() == chunk_size, "Invalid MvccData size");
-  for (const auto& segment : segments) {
+  for (const auto& segment : _segments) {
+    Assert(!mvcc_data || !std::dynamic_pointer_cast<ReferenceSegment>(segment), "Chunks containing ReferenceSegments should not contains MvccData. They implicitly use the MvccData of the referenced Table");
```

Comment:
This `Assert` previously was actually present in Release builds, now it's not anymore because it is within `HYRISE_DEBUG`. Is that intended?

Reply:
I think it's fine to only check this in debug builds as it's unlikely user input could ever trigger this when tests (running in debug mode) wouldn't. But strong conviction on my part.